cpp/include/raft/core/kvp.hpp

```diff
@@ -20,7 +20,7 @@
 
 #ifdef _RAFT_HAS_CUDA
 #include <cub/cub.cuh>
-#include <raft/util/cuda_utils.cuh>
+#include <raft/util/cuda_utils.cuh>  // raft::shfl_xor
```

Comment:
I believe RAFT's developer guide makes no recommendations to have comments for headers that are being included, and as such I'd like to avoid them completely since the rest of the codebase doesn't use them

Reply:
I really needed these comments while developing this PR though. It is so easy to leave an unneeded header on a file. Even though I was actively trying to reduce the number of includes, I constantly found unneeded includes that I had inadvertently left in. With these comments, you can at least quickly search through the file if for instance `raft::shfl_xor` is still used. If not, it might be worth trying to remove the include. If it is, the check was quick and easy.

Would it be okay to go with the following policy: 
- leave the comments when they are already present
- update them when they are present
- not require them on new code/code that doesn't have these comments yet

I agree that consistency should be valued, but I also think that removing these comments in the entire distance/detail tree would really make it hard to keep the includes minimal (especially during review).